jdaviz/core/tools.py

```diff
@@ -249,3 +249,39 @@ class SidebarShortcutCompass(_BaseSidebarShortcut):
     tool_id = 'jdaviz:sidebar_compass'
     action_text = 'Compass'
     tool_tip = 'Open compass plugin in sidebar'
+
+
+@viewer_tool
+class SinglePixelRegion(CheckableTool):
+
+    icon = os.path.join(ICON_DIR, 'pixelspectra.svg')  # FIXME: Update icon
+    tool_id = 'jdaviz:singlepixelregion'
+    action_text = 'Create single-pixel spatial region'
+    tool_tip = 'Click on the viewer to create single-pixel spatial region'
+
+    def activate(self):
+        self.viewer.add_event_callback(self.on_mouse_event, events=['click'])
+
+    def deactivate(self):
+        self.viewer.remove_event_callback(self.on_mouse_event)
+
+    def on_mouse_event(self, data):
+        # Extract data coordinates - these are pixels in the reference image.
+        # NOTE: We always use the reference image pixel coordinates because
+        # Subset is defined w.r.t. reference image.
+        x = data['domain']['x']
+        y = data['domain']['y']
+        reg = self.get_subset(x, y, as_roi=False)
+        self.viewer.jdaviz_helper.load_regions(reg)
```

Comment:
this ignores the subset edit mode and seems to always reset to "create new subset".  Is that the intended behavior?  I'd expect to be able to build up a single subset of multiple pixels with "add" logic, or remove single pixels from a circle subset with "remove", etc.

Reply:
This might be the difference between `load_regions` (which forces loading a _new_ region) and `apply_roi` (which adheres to the currently selected mode)?

Reply:
Yes, it is on purpose. If you "build up a single subset of multiple pixels," you are basically manually drawing a masked Subset, which is pretty much useless for anything because we do not have a translator for it.

Reply:
Unless that is what @camipacifici wants ❓ but you cannot get back a proper Region object. 🤷

Reply:
is that the same limitation for all the rest of the subsets once combining with the various operators?  If so, I think that's fine - allow extracting a region when possible, but at least they all act the same and can be used as subsets in any of the plugins.  Otherwise we somehow need to make a distinction in the behavior between regions and subsets more clear to the user.

Reply:
@pllim, you are correct. It is important to have the ability to use these subsets elsewhere, so "multiple individual subsets" is ok.

Reply:
If you mean compound Subset, that is pretty useless too (also resolves to MaskedSubset state, I think?). You cannot edit that after creation.

Reply:
Ok, then I think it would be consistent to do the same here.  A single pixel can be extracted as a region.  If the user combines to create a compound Subset, then it comes with the same limitations as any other compound subset.  If they want multiple individual single pixel subsets, they can do that in the same way as they would multiple individual circle subsets (by either toggling the tool or clicking the "create new subset" from the app toolbar).

Reply:
Okay, I make them the same now.